src/main/java/org/terasology/launcher/jre/ManagedJRE.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.launcher.jre;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.terasology.launcher.platform.Arch;
+import org.terasology.launcher.platform.OS;
+import org.terasology.launcher.platform.Platform;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public final class ManagedJRE {
+
+    public static final Set<Integer> SUPPORTED_JAVA_VERSIONS = Sets.newHashSet(8, 11, 17);
+
+    private static final Set<JREArtefact> supportedJavaVersions;
```

Comment:
Hm, bad naming 🤔 `SUPPORTED_JAVA_VERSIONS` denotes all the major Java versions we may need, while `supportedJavaVersions` is actually something like `jreInfoForAllSupportedPlatformsAndJavaVersions` - what a mouthful!